### PR TITLE
Add new filetype for source code files and icons

### PR DESCRIPTION
**Pull request by cfxegbert**

### CODE REVIEW

File: src/theme/default_theme.rs [hunk 1]
```diff
@@ -104,6 +104,7 @@ impl UiStyles {
                 temp:       White.normal(),
                 compiled:   Yellow.normal(),
                 build:      Yellow.bold().underline(),
+                source:     Yellow.bold(), // Need to discuss color
```
Comment (MartinFillon) [inline src/theme/default_theme.rs:107]: I think yellow bold is already somewhere, but if not its good if people are not happy they will change it

**Review by MartinFillon — APPROVED** [review]

File: src/theme/default_theme.rs (hunk 1 above)
Comment (MartinFillon) [reply src/theme/default_theme.rs:107]: Yeah I just saw the problem is it is to near the build one

**Review by MartinFillon — CHANGES_REQUESTED** [review]
see comment right above

File: src/theme/default_theme.rs (hunk 1 above)
Comment (cfxegbert) [reply src/theme/default_theme.rs:107]: That was sort of intentional.  Build scripts/config or a sort of source code also.

Comment (MartinFillon) [reply src/theme/default_theme.rs:107]: Yeah but I think that even tho they are sort of source code and are useful all together, but I think that if @cafkafk is okay with that we just keep it and I will change my eza colors

**Comment by cfxegbert** [comment]
https://gist.github.com/cfxegbert/77c32021d73e006c1c5b86419548ff6e

# Added extensions
| Ext         | Icon |          Class          |  Code   | Notes                                      |
|-------------|:----:|:-----------------------:|:-------:|--------------------------------------------|
| aif         |     |       nf-fa-music       | \uf001  |                                            |
| aifc        |     |       nf-fa-music       | \uf001  |                                            |
| aiff        |     |       nf-fa-music       | \uf001  |                                            |
| arj         |     |     nf-oct-file_zip     | \uf410  |                                            |
| cr          |     |     nf-seti-crystal     | \ue62f  | Crystal source code file                   |
| crdownload  |  󰇚  |     nf-md-download      | \uf01da | Chrome download temp file                  |
| d           |     |      nf-dev-dlang       | \ue7af  | DLang source code file                     |
| di          |     |      nf-dev-dlang       | \ue7af  | DLang interface file                       |
| fnt         |     |       nf-fa-font        | \uf031  | Windows font file                          |
| fon         |     |       nf-fa-font        | \uf031  | Windows font library                       |
| gvy         |     |      nf-dev-groovy      | \ue775  | Groovy source code file                    |
| h++         |     |      nf-custom-cpp      | \ue61d  | C++ header file                            |
| h264        |     |   nf-fa-video_camera    | \uf03d  |                                            |
| haml        |     |      nf-seti-haml       | \ue664  | Haml markup file                           |
| inl         |     |       nf-custom-c       | \ue61e  | Inline source sometimes used with C or C++ |
| ltx         |     |       nf-seti-tex       | \ue69b  | LaTex source                               |
| mid         |  󰣲  |     nf-md-midi_port     | \uf08f2 | Midi file                                  |
| pcm         |     |       nf-fa-music       | \uf001  |                                            |
| pps         |     | nf-fa-file_powerpoint_o | \uf1c4  |                                            |
| ppsx        |     | nf-fa-file_powerpoint_o | \uf1c4  |                                            |
| purs        |     |   nf-seti-purescript    | \ue630  | PureScript                                 |
| sbt         |  󰨖  |     nf-md-subtitles     | \uf0a16 |                                            |
| shtml       |     |       nf-fa-html5       | \uf13b  |                                            |
| ssa         |  󰨖  |     nf-md-subtitles     | \uf0a16 |                                            |
| tml         |     |     nf-seti-config      | \ue615  | Short form of .toml extension              |
| v           |     |    nf-custom-v_lang     | \ue6ac  | V language source code                     |
| webmanifest |     |      nf-seti-json       | \ue60b  |                                            |
| wv          |     |       nf-fa-music       | \uf001  |                                            |
| xlr         |     | nf-fa-file_powerpoint_o | \uf1c4  |                                            |


# Changed extensions
| Ext  | Icon |     Class      |  Code   | Old icon |  Old class   | Old code | Notes                                             |
|------|:----:|:--------------:|:-------:|:--------:|:------------:|:--------:|---------------------------------------------------|
| part |  󰇚  | nf-md-download | \uf01da |         | nf-oct-clock |  \uf43a  | Partial download                                  |
| zig  |     |  nf-seti-zig   | \ue6a9  |    ↯     |              |  \u21af  | Official zig icon.  Old icon was not in Nerdfonts |

# Removed extensions
| Ext | Notes |
|-----|-------|

# Added files
| File           | Icon |     Class      |  Code  | Notes                      |
|----------------|:----:|:--------------:|:------:|----------------------------|
| .eslintrc.cjs  |     | nf-seti-eslint | \ue655 |                            |
| .eslintrc.js   |     | nf-seti-eslint | \ue655 |                            |
| .eslintrc.json |     | nf-seti-eslint | \ue655 |                            |
| .eslintrc.yaml |     | nf-seti-eslint | \ue655 |                            |
| .eslintrc.yml  |     | nf-seti-eslint | \ue655 |                            |
| .mailmap       |     |   nf-fa-git    | \uf1d3 |                            |
| COPYING        |     |   nf-fa-book   | \uf02d | Sometimes contains license |
| COPYRIGHT      |     |   nf-fa-book   | \uf02d | Sometimes contains license |
| LICENSE-APACHE |     |   nf-fa-book   | \uf02d |                            |
| LICENSE-MIT    |     |   nf-fa-book   | \uf02d |                            |



# Changed files
| File | Icon | Class | Code | Old icon | Old class | Old code | Notes |
|------|:----:|:-----:|:----:|:--------:|:---------:|:--------:|-------|

# Removed files
| File | Notes |
|------|-------|

**Review by cafkafk — APPROVED** [review]
LGTM, tests pass and changes make sense

**Pull request merged**